src/coreclr/nativeaot/System.Private.CoreLib/src/System/Collections/Generic/EqualityComparer.NativeAot.cs

```diff
@@ -23,6 +23,12 @@ private static EqualityComparer<T> Create()
             // This body serves as a fallback when instantiation-specific implementation is unavailable.
             // If that happens, the compiler ensures we generate data structures to make the fallback work
             // when this method is compiled.
+
+            if (typeof(T) == typeof(string))
+            {
+                return Unsafe.As<EqualityComparer<T>>(new StringEqualityComparer());
```

Comment:
This Unsafe.As isn't necessary. I'll wait to clean it up until I see what happens with the current CI, and either fix it as part of another round of fixes or in a follow-up.